runtime/src/main/java/org/corfudb/protocols/CorfuProtocolCommon.java

```diff
@@ -255,6 +257,28 @@ public static StreamAddressRangeMsg getStreamAddressRangeMsg(StreamAddressRange
                 .build();
     }
 
+    /**
+     * Returns a StreamAddressResponse object from its log tail, epoch, and List
+     * of address map entries, each consisting of a UUID and a StreamAddressSpace,
+     * represented in Protobuf.
+     *
+     * @param tail   the log tail
+     * @param epoch  the epoch the response was sealed with
+     * @param map    a list of address map entries represented in Protobuf
+     * @return       an equivalent StreamsAddressResponse object
+     */
+    public static StreamsAddressResponse getStreamsAddressResponse(long tail, long epoch,
+                                                                   List<UuidToStreamAddressSpacePairMsg> map) {
+        StreamsAddressResponse response = new StreamsAddressResponse(tail,
+                map.stream().collect(Collectors.<UuidToStreamAddressSpacePairMsg, UUID, StreamAddressSpace>toMap(
+                        entry -> getUUID(entry.getStreamUuid()),
+                        entry -> getStreamAddressSpace(entry.getAddressSpace())
+                )));
+
+        response.setEpoch(epoch);
+        return response;
+    }
+
```

Comment:
There's actually an identical method in `CorfuProtocolSequencer`. My original intent was to place this method in `CorfuProtocolCommon` since it is used by both the `Sequencer` and `LogUnit`, but I must have missed this. Given that this can be used by both servers, I think it would be better to leave the implementation in `CorfuProtocolCommon` and remove the references to the one in `CorfuProtocolSequencer`? This would avoid duplication.